### DISCUSSION TITLE
Tabview tab sorting

### DISCUSSION THREAD

**Discussion by howroyd** [TomSchimansky/CustomTkinter#2560] [Q&A]
Is there a way to sort tabs in a tabview once they have been added?

In detail with context:
I am adding tabs to a tabview at runtime based upon incoming data.  `tabview.add()` works fine, however I would like to re-order these tabs in lexographical order.  From what I can see in the docs, this cannot be done?
My workaround is to store the data to make each tab externally, append to that when new data comes in, sort it, then `delete()` and `add()` in the new sorted order.  This is obviously cumbersome and an interesting side effect is that the _in focus_ tab is becomes blank until it is manually clicked on again.

I can make a minimum working example if required, but in non optimal, psuedo-code to show the general workflow in case I'm missing an obvious:

```python3
tabview = ctk.CTkTabview(root)
tabs: dict[str, MyTabClass] = {}

while True:
  new_data = queue.get()

  if new_data.key not in tabs:
    tabs[new_data.key] = make_new_tab(new_data)

  for k in tabs.keys():
    tabview.delete(k)

  tabs = dict(sorted(tabs.items()))

  for k, v in tabs.items():
    remake_tab(tabview, k, new_data)

  root.update()
```

**Comment by dipeshSam** [comment]
I've created a function `sort_tabs` for you. Here is your solution:
``` Python
from customtkinter import CTk, CTkTabview, CTkLabel 
 

def sort_tabs(tabs: CTkTabview):    
    """Sorts all the tabs of `CTkTabView` in lexicographical order."""
    
    for i, sorted_name in enumerate(sorted(tabs._name_list)):
         tabs._segmented_button.move(i, sorted_name)     
                         
    tabs.update()
   

if __name__ == "__main__":
    app = CTk()

    tabs = CTkTabview(app, width=400) 
    tabs.pack(padx=20, pady=20)

    mango  = tabs.add("Mango")
    apple  =  tabs.add("Apple")
    orange = tabs.add("Orange")
    banana = tabs.add("Banana")
    grape  = tabs.add("Grape")
    
    label_font = ("Arial", 30)
    CTkLabel(mango, text="This is Mango tab.", font  =  label_font).place(relx=0.5, anchor="center", rely=0.5)
    CTkLabel(apple, text="This is Apple tab.", font  =  label_font).place(relx=0.5, anchor="center", rely=0.5)
    CTkLabel(orange, text="This is Orange tab.", font = label_font).place(relx=0.5, anchor="center", rely=0.5)
    CTkLabel(banana, text="This is Banana tab.", font = label_font).place(relx=0.5, anchor="center", rely=0.5)
    CTkLabel(grape, text="This is Grape tab.", font  =  label_font).place(relx=0.5, anchor="center", rely=0.5)
    
    
    # Calling sorting method
    sort_tabs(tabs)
    
    
    app.mainloop()
```

Before: 
![tabview_sort_BEFORE](https://github.com/user-attachments/assets/b5105130-e5ec-4441-b4c7-6dff80c1f3fb)

**After:** 
![tabview_sort_AFTER](https://github.com/user-attachments/assets/1a5199d0-d62e-4802-a108-5bfef95eb409)

Hope this would be helpful for you.
Regards.
